pipe.c

```diff
@@ -47,30 +82,51 @@ main(int argc, char *argv[]) {
                 perror("read");
                 return 1;
             }
-
+	    if (i == buf->seq) {
+                gettimeofday(&tv_post_read, NULL);
+                time_elapsed = time_diff(buf->tv_time_sent, tv_post_read);
+		latencies[i]= time_elapsed;
+                tot_elapsed+=time_elapsed;
+            }
+            else {
+                    printf("We did not match a sequence.\n");
+                    exit(1);
+            }
             sum += n;
         }
 
         if (sum != count * size) {
-            fprintf(stderr, "sum error: %d != %d\n", sum, count * size);
+            fprintf(stderr, "sum error: %ld != %ld\n", sum, count * size);
             return 1;
         }
+	avg_elapsed = tot_elapsed/count;
+	ninetyninth=(count*0.99)-1;
+        ninetyfifth=(count*0.95)-1;
+        qsort( latencies, count, sizeof(double), compare_double );
+
+        printf("%.0fusMIN\n", latencies[0]);
+        printf("%.0fusAVG\n", avg_elapsed);
+	printf("%.0fusMAX\n", latencies[count-1]);
+	printf("%.0fus95th\n", latencies[ninetyfifth]);
+        printf("%.0fus99th\n", latencies[ninetyninth]);
```

Comment:
I don't think this format of cramming the units up against the value does us any favors for reporting or post-processing. I think we need to consider a key:value or a tabular format.